emsdk.py

```diff
@@ -838,6 +838,7 @@ def git_checkout_and_pull(repo_path, branch_or_tag):
       ret = run([GIT(), 'merge', '--ff-only', 'origin/' + branch_or_tag], repo_path)
     if ret != 0:
       return False
+    run([GIT(), 'submodule', 'update', '--init'], repo_path, quiet=True)
```

Comment:
Alternatively, we could add `--recurse-submodules` (and conditionally `--shallow-submodules`) to the clone step. If not, we should probably add `--recursive` and conditionally `--depth=1` if `GIT_CLONE_SHALLOW` here.

Reply:
Good idea.  I added `--recurse-submodules` to both clone and checkout step.  Sadly the GIT_CLONE_SHALLOW option is not tested anywhere.. and even building from source at all is barely tested.